packages/db/src/runtime/db-client.ts

```diff
@@ -36,6 +38,44 @@ const remoteResultSchema = z.object({
 });
 
 export function createRemoteDatabaseClient(appToken: string, remoteDbURL: string) {
+	const remoteUrl = new URL(remoteDbURL);
+
+	switch (remoteUrl.protocol) {
+		case 'http:':
+		case 'https:':
+			return createStudioDatabaseClient(appToken, remoteUrl);
+		case 'libsql+http:':
+			remoteUrl.protocol = 'http:';
+			return createRemoteLibSQLClient(appToken, remoteUrl);
+		case 'libsql+https:':
+			remoteUrl.protocol = 'https:';
+			return createRemoteLibSQLClient(appToken, remoteUrl);
+		case 'ws:':
+		case 'wss:':
+		case 'file:':
+		case 'memory:':
+		case 'libsql:':
+			return createRemoteLibSQLClient(appToken, remoteUrl);
+		default:
+			throw new Error(`Unsupported remote DB: ${remoteDbURL}`);
+	}
+}
+
+function createRemoteLibSQLClient(appToken: string, remoteDbURL: URL) {
+	const options: Partial<LibSQLConfig> = Object.fromEntries(remoteDbURL.searchParams.entries());
+	remoteDbURL.search = '';
+
+	const client = createClient({
+		...options,
+		authToken: appToken,
+		url: remoteDbURL.protocol === 'memory:' ? ':memory:' : remoteDbURL.toString(),
```

Comment:
It's a bit weird to check for `memory:` **again**, since we already do that already inside `createRemoteDatabaseClient`, in the `switch` statement you added. I wonder if we can do the check just once.